compiler/rustc_smir/src/stable_mir/ty.rs

```diff
@@ -1,6 +1,6 @@
 use super::with;
 
-#[derive(Copy, Clone, Debug)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
```

Comment:
Comparisons of types are wrong. They compare just the index, which has no meaning